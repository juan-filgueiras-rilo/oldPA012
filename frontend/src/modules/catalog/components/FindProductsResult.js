import React from 'react';
import {connect} from 'react-redux';
import {FormattedMessage} from 'react-intl';

import * as selectors from '../selectors';
import {Pager} from '../../common';
import Products from './Products';

export const FindProductsResult = ({productSearch, categories, history, search}) => {

    if (!productSearch) {
        return null;
    }

    if (productSearch.result.items.length === 0) {
        return (
            <div className="alert alert-danger" role="alert">
                <FormattedMessage id='project.catalog.FindProductsResult.noProductsFound'/>
            </div>
        );
    }

    return (
        
        <div>
            <Products products={productSearch.result.items} categories={categories}/>
            <Pager 
                back={{
                    enabled: productSearch.criteria.page >= 1,
                    onClick: () => {
                        const newParams = new URLSearchParams(search);
                        newParams.set('page', Number(newParams.get('page'))-1);
                        history.push(`/catalog/find-products/search?`+newParams);
                    }
                }}
                next={{
                    enabled: productSearch.result.existMoreItems,
                    onClick: () => {
                        const newParams = new URLSearchParams(search);
                        newParams.set('page', Number(newParams.get('page'))+1);
                        history.push(`/catalog/find-products/search?`+newParams);
                    }
                }}
            />
        </div>

    );

}

const mapStateToProps = (state) => ({
    productSearch: selectors.getProductSearch(state),
    categories: selectors.getCategories(state)
});

export default connect(mapStateToProps)(FindProductsResult);