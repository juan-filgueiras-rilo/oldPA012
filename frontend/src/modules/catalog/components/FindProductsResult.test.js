import React from 'react';
import { FindProductsResult } from './FindProductsResult';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { initReactIntl } from '../../../i18n';

const {locale, messages} = initReactIntl();

jest.mock ('./Products', () => 'Products')
jest.mock ('../../common/components/Pager', () => 'Pager')

const emptyProductSearch = {
    criteria: null,
    result: {
        items: [],
        existMoreItems: false
    }
};

const productSearch = {
    criteria: {
        keywords: "",
        categoryId: "",
        page: 0,
    },
    result: {
        items: [
            {
                "id": 2,
                "categoryId": 1,
                "name": "producto2",
                "currentPrice": 10,
                "remainingTime": 993853
            },
            {
                "id": 1,
                "categoryId": 2,
                "name": "producto1",
                "currentPrice": 12.5,
                "remainingTime": 10
            }
        ],
        existMoreItems: false
    },
};

const categories = ["Movies","Technology"];

it('renders error message when there is no products', () => {
    const tree = renderer
    .create(
        <IntlProvider locale={locale} messages={messages}>
            <FindProductsResult productSearch={emptyProductSearch} categories={categories}/>
        </IntlProvider>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders products and pager when there is at least one product', () => {
    const tree = renderer
    .create(
        <FindProductsResult productSearch={productSearch} categories={categories}/>
    )
    .toJSON();
    expect(tree).toMatchSnapshot();
});