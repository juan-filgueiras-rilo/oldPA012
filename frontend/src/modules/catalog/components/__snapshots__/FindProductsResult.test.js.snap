// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error message when there is no products 1`] = `
<div
  className="alert alert-danger"
  role="alert"
>
  <span>
    project.catalog.FindProductsResult.noProductsFound
  </span>
</div>
`;
